api/tests/integration/domain/services/smart-random/smart-random_test.js

```diff
@@ -561,13 +629,19 @@ describe('Integration | Domain | Stategies | SmartRandom', () => {
         ];
 
         // when
-        const { nextChallenge } = SmartRandom.getNextChallenge({ targetSkills, challenges, knowledgeElements, lastAnswer });
+        const { nextChallenge } = SmartRandom.getNextChallenge({
+          targetSkills,
+          challenges,
+          knowledgeElements,
+          lastAnswer,
+          assessmentId
+        });
 
         // then
         expect(nextChallenge).to.be.equal(null);
       });
 
-      it('should select a challenge randomly among challenges of equal reward', function() {
+      it('should select a challenge among challenges of equal reward', function() {
```

Comment:
Je retranscrirai bien l'intention de la sélection déterministe dans le nom du test.
Par exemple avec `'should do a deterministic selection of a challenge among challenges of equal reward '`

Reply:
+1